byway/src/main/java/com/google/sps/servlets/PlacesServlet.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.maps;
+
+import com.google.maps.model.LatLng;
+import com.google.maps.model.PlaceType;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns all PlaceTypes from Places API. */
+@WebServlet("/places")
+public class PlacesServlet extends HttpServlet {
+
+  private final Gson gson = new Gson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> places = new ArrayList<>();
+    for(PlaceType location: PlaceType.values()) {
+      String place = formatLocation(location.toString());
+      places.add(place.toString());
+    }
+    response.setContentType("text/html;");
+    response.getWriter().println(gson.toJson(places));
+  }
+
+  /** 
+   * Format string by capitalizing and adding spaces in-between words.
+   * @param place is the location name.
+   */
+  private String formatLocation(String place) {
```

Comment:
If the argument is supposed to be a `PlaceType`, it would be better to be explicit about that by taking a `PlaceType` as the argument rather than a `String`.

Reply:
Fixed